src/s3/util/fileViewerManager.ts

```diff
@@ -55,13 +65,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        //TODOD:: delegate this logic to S3Tab.ts
-        //this will display the document at the end
-        this.window.showTextDocument(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        await newTab.openFileOnReadOnly()
+
+        this.activeTabs.set(fileLocation.fsPath, newTab)
+    }
+
+    public async openOnEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (uriOrNode instanceof vscode.Uri) {
+            //was activated from an open tab
+            if (this.activeTabs.has(uriOrNode.fsPath)) {
+                const tab = this.activeTabs.get(uriOrNode.fsPath)
+                await tab!.openFileOnEditMode()
+            }
+        } else if (uriOrNode instanceof S3FileNode) {
+            //was activated from the explorer, need to get the file
+            const fileLocation = await this.getFile(uriOrNode)
+            if (!fileLocation) {
+                return
```

Comment:
This should be accompanied by some sort of messaging, probably a log statement and a warning message (I'm assuming this isn't an expected case?)

Reply:
So the log messages happen right before it returns `undefined` it can either be because user cancelled because of one of the prompts(>4MB and !fileNode.file.sizeBytes), or the folder structure creation fails (which logs the failure), or the download fails (which outputs an error message)